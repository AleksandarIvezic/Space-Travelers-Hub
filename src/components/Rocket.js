import React from 'react';
import PropTypes from 'prop-types';
import Button from 'react-bootstrap/Button';

const Rocket = ({ name, image, description }) => (
  <li className="d-flex m-2 p-2">
    <div className="image mx-2">
      <img src={image} alt="rocket-img" width="250" height="180" />
    </div>
    <div className="text">
      <h3>{name}</h3>
      <p>{description}</p>
      <Button type="button" className="btn btn-primary">Reserve rocket</Button>
    </div>
  </li>
);

export default Rocket;

Rocket.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
};
