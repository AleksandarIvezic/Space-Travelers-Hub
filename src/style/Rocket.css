.btn-reserved {
  background-color: transparent;
  border-color: #6d757d;
  color: #6d757d;
  display: inline-block;
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  border-radius: 0.25rem;
}
