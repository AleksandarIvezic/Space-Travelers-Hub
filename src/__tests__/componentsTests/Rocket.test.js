import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../redux/storeConfig';
import Rocket from '../../components/Rocket';

describe('Test if Rocket', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Rocket
            key={1}
            name="Rocket name"
            image="image link"
            description="rocket description"
            id={1}
          />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
