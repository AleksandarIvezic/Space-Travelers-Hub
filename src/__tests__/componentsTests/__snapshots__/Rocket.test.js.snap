// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if Rocket renders correctly 1`] = `
<li
  className="d-flex flex-wrap flex-lg-nowrap m-2 p-2"
>
  <div
    className="image mx-2"
  >
    <img
      alt="rocket-img"
      height="180"
      src="image link"
      width="250"
    />
  </div>
  <div
    className="text"
  >
    <h3>
      Rocket name
    </h3>
    <p>
      rocket description
    </p>
    <button
      className="btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Reserve Rocket
    </button>
  </div>
</li>
`;
